corehq/apps/userreports/models.py

```diff
@@ -399,11 +402,17 @@ def named_expression_objects(self):
     @property
     @memoized
     def named_filter_objects(self):
-        return {name: FilterFactory.from_spec(filter, FactoryContext(self.named_expression_objects, {}))
-                for name, filter in self.named_filters.items()}
+        return {
+            name: FilterFactory.from_spec(
+                filter, FactoryContext(
+                    self.named_expression_objects, {}, domain=self.domain
+                )
```

Comment:
You should move the factory context creation to outside the loop otherwise the context caching will be of no value.

Reply:
https://github.com/dimagi/commcare-hq/pull/31443/commits/4a986c123fea809e53ac71a7861c6e5fe64ee79a